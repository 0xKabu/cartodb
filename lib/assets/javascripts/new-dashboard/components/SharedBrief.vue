<template>
  <div>
    <img class="brief__icon" svg-inline src="../assets/icons/common/user.svg">
    <span class="brief__text text is-small is-txtSoftGrey">
      {{ $t(`SharedBrief.sharedWith`) }} {{ isSharedWithOrg ? $tc(`SharedBrief.org`) : sharedWithText }}
    </span>
  </div>
</template>

<script>
export default {
  name: 'SharedBrief',
  props: {
    colleagues: Array
  },
  computed: {
    isSharedWithOrg () {
      return this.colleagues.some(colleague => colleague.type === 'org');
    },
    isSharedWithGroups () {
      return this.groups.length > 0;
    },
    isSharedWithUsers () {
      return this.users.length > 0;
    },
    groups () {
      return this.colleagues.filter(colleague => colleague.type === 'group');
    },
    users () {
      return this.colleagues.filter(colleague => colleague.type === 'user');
    },
    userText () {
      return this.$tc(`SharedBrief.users`, this.users.length, { username: this.users[0].entity.username });
    },
    groupText () {
      return this.$tc(`SharedBrief.groups`, this.groups.length, { groupname: this.groups[0].entity.name });
    },
    sharedWithText () {
      const andText = this.$t(`SharedBrief.and`);
      if (this.isSharedWithUsers && this.isSharedWithGroups) {
        return `${this.userText} ${andText} ${this.groupText}`;
      } else if (this.isSharedWithUsers) {
        return `${this.userText}`;
      } else if (this.isSharedWithGroups) {
        return `${this.groupText}`;
      }
    }
  }
};
</script>

<style lang="scss" scoped>
@import 'new-dashboard/styles/variables';

.brief__icon {
  margin-right: 4px;
}

.brief__text {
  line-height: 15px;
  white-space: nowrap;
}

</style>
