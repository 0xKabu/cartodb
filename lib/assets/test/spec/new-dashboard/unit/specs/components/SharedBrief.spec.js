import { shallowMount } from '@vue/test-utils';
import SharedBrief from 'new-dashboard/components/SharedBrief';

const $tc = key => key;
const $t = key => key;

describe('SharedBrief.vue', () => {
  it('should render correct contents', () => {
    const colleaguesData = [
      { type: 'group',
        entity: {
          id: 'xxxx-yyy',
          name: 'grupoprueba'
        },
        access: 'r' },
      { type: 'user',
        entity: {
          id: 'zzzz-www',
          username: 'user1'
        },
        access: 'w'
      }
    ];
    const sharedBrief = shallowMount(SharedBrief, {
      propsData: {
        colleagues: colleaguesData
      },
      mocks: {
        $tc, $t
      }
    });

    expect(sharedBrief).toMatchSnapshot();
  });
});
