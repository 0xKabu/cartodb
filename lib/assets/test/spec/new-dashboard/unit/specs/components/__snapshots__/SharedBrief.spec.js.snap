// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SharedBrief.vue should render the text for groups 1`] = `
<div><img svg-inline="" src="../assets/icons/common/user.svg" class="brief__icon"> <span class="brief__text text is-small is-txtSoftGrey">
    SharedBrief.sharedWith SharedBrief.groups
  </span></div>
`;

exports[`SharedBrief.vue should render the text for groups and users 1`] = `
<div><img svg-inline="" src="../assets/icons/common/user.svg" class="brief__icon"> <span class="brief__text text is-small is-txtSoftGrey">
    SharedBrief.sharedWith SharedBrief.users SharedBrief.and SharedBrief.groups
  </span></div>
`;

exports[`SharedBrief.vue should render the text for sharing with all organization members 1`] = `
<div><img svg-inline="" src="../assets/icons/common/user.svg" class="brief__icon"> <span class="brief__text text is-small is-txtSoftGrey">
    SharedBrief.sharedWith SharedBrief.org
  </span></div>
`;

exports[`SharedBrief.vue should render the text for users 1`] = `
<div><img svg-inline="" src="../assets/icons/common/user.svg" class="brief__icon"> <span class="brief__text text is-small is-txtSoftGrey">
    SharedBrief.sharedWith SharedBrief.users
  </span></div>
`;
